builtin/providers/azurerm/resource_arm_storage_account.go

```diff
@@ -153,6 +181,17 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 		},
 	}
 
+	// AccessTier is only valid for BlobStorage accounts
+	if accountKind == string(storage.BlobStorage) {
+		accessTier, ok := d.GetOk("access_tier")
```

Comment:
I think we should add the default as part of the scheme declaration rather than defaulting in code